vncviewer/EmulateMB.cxx

```diff
@@ -194,23 +205,198 @@ static const signed char stateTab[11][5][3] = {
   },
 };
 
-EmulateMB::EmulateMB()
-  : state(0), emulatedButtonMask(0), timer(this)
+/*
+ * 2) Pressing a modifier + left button to emulate middle button:
+ *
+ * At each state, we need handlers for the following events
+ *   0: Modifier pressed
+ *   1: Modifier released
+ *   2: Left button pressed / released
+ *   3: Other button pressed / released
```

Comment:
Unfortunately multiple buttons can be pressed/released in the same event. So things need to be structured like the existing state table.

I'm not sure, but it might also be an advantage to change the keyboard handling to the same line of thinking. I.e. track what state the keys are in, rather than the change. It might be confusing that for mouse we react to current state, but for keys we react to the change.